### ISSUE TITLE
Adding A11y Storybook addon.

### ISSUE THREAD

**Issue by Tribunal33** [pkp/pkp-lib#10893]
## Issue Description
Adding an addon for storybook.
Automatically scans for accessibility violations within storybook.  [Ally Plugin tool](https://storybook.js.org/addons/@storybook/addon-a11y).

As part of our ORE responsibilities this will help ensure Accessibility is being addressed further left into the development process.  

Installation process: 
in lib/ui-library run : `npm install -D @storybook/addon-a11y@8.0.5`

Then add this to lib\ui-library\.storybook\main.js in the addons section
'@storybook/addon-a11y',

Note : If you install it out of the box it tries to upgrade storybook to 8.5.3 there are incompatibility issues since we are using some older version of other addons.  
You can upgrade using `npm install @storybook/addon-a11y@8.5.3 @storybook/test@8.5.3 @storybook/addon-essentials@8.5.3 @storybook/addon-interactions@8.5.3 @storybook/addon-mdx-gfm@8.5.3 @storybook/blocks@8.5.3 @storybook/vue3@8.5.3 @storybook/vue3-vite@8.5.3` but we are not trying to upgarde storybook core at this time.

To use
When opening up a component in storybook there should be a new tab at the bottom (see screenshot).  This will highlight any accessibility issues.

![Image](https://github.com/user-attachments/assets/673c7f66-a726-4c55-a97e-c95c2056f8a8)

## Additional information
Additionally upgrade storybook to latest 8.5.3 `npx storybook@latest upgrade` and all dependencies -- Including suggested additional upgrade to latest for caniuse-lite.

- eslint-plugin-storybook: 0.8.0 => 0.11.2 │
- msw-storybook-addon: 2.0.0-beta.1 => 2.0.4 
- storybook-mock-date-decorator: 1.0.2 => 2.0.6

PR:
https://github.com/pkp/ui-library/pull/517

**Comment by Tribunal33** [comment]
PR has merged
